File: src/main/java/me/lokka30/levelledmobs/misc/QueueItem.java
/*
 * Copyright (c) 2020-2021  lokka30. Use of this source code is governed by the GNU AGPL v3.0 license that can be found in the LICENSE.md file.
 */

package me.lokka30.levelledmobs.misc;

import org.bukkit.entity.Player;
import org.bukkit.event.Event;

import java.util.List;

/**
 * Holds data that goes into various queues for processing
 *
 * @author stumper66
 * @since 3.0.0
 */
public class QueueItem {
    public QueueItem(final LivingEntityWrapper lmEntity, final Event event){
        this.lmEntity = lmEntity;
        this.event = event;
    }

    public QueueItem(final LivingEntityWrapper lmEntity, final String nametag, final List<Player> players){
        this.lmEntity = lmEntity;
        this.nametag = nametag;
        this.players = players;
    }

    public final LivingEntityWrapper lmEntity;
    public Event event;
    public List<Player> players;
    public String nametag;
}

File: levelledmobs-plugin/src/main/kotlin/io/github/arcaneplugins/levelledmobs/misc/QueueItem.kt
package io.github.arcaneplugins.levelledmobs.misc

import io.github.arcaneplugins.levelledmobs.result.NametagResult
import io.github.arcaneplugins.levelledmobs.wrappers.LivingEntityWrapper
import java.util.UUID
import org.bukkit.entity.Player
import org.bukkit.event.Event

/**
 * Holds data that goes into various queues for processing
 *
 * @author stumper66
 * @since 3.0.0
 */
class QueueItem {
    var entityId: UUID
        private set
    var lmEntity: LivingEntityWrapper
        private set
    var event: Event? = null
        private set
    var players: MutableList<Player>? = null
    var nametag: NametagResult? = null

    constructor(
        lmEntity: LivingEntityWrapper,
        event: Event?
    ){
        this.lmEntity = lmEntity
        this.event = event
        this.entityId = lmEntity.livingEntity.uniqueId
    }

    constructor(
        lmEntity: LivingEntityWrapper,
        nametag: NametagResult,
        players: MutableList<Player>
    ){
        this.lmEntity = lmEntity
        this.nametag = nametag
        this.players = players
        this.entityId = lmEntity.livingEntity.uniqueId
    }
}